cmd/samples/complex/sender/main.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/client"
+	cloudeventshttp "github.com/cloudevents/sdk-go/pkg/cloudevents/transport/http"
+	cloudeventsnats "github.com/cloudevents/sdk-go/pkg/cloudevents/transport/nats"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/types"
+	"github.com/google/uuid"
+	"github.com/kelseyhightower/envconfig"
+	"log"
+	"net/url"
+	"os"
+	"time"
+)
+
+const (
+	count = 1
+	delay = 100 * time.Millisecond
+)
+
+type envConfig struct {
+	// Target URL where to send cloudevents
+	HttpTarget string `envconfig:"HTTP_TARGET" default:"http://localhost:8080" required:"true"`
+
+	// NatsServer URL to connect to the nats server.
+	NatsServer string `envconfig:"NATS_SERVER" default:"http://localhost:4222" required:"true"`
+
+	// Subject is the nats subject to publish cloudevents on.
+	Subject string `envconfig:"SUBJECT" default:"sample" required:"true"`
+}
+
+func main() {
+	var env envConfig
+	if err := envconfig.Process("", &env); err != nil {
+		log.Printf("[ERROR] Failed to process env var: %s", err)
+		os.Exit(1)
+	}
+	os.Exit(_main(os.Args[1:], env))
+}
+
+// Simple holder for the sending sample.
+type Demo struct {
+	Client *client.Client
+
+	// Content
+	EventType   string
+	Source      url.URL
+	ContentType string
+
+	// Data
+	Message string
+
+	seq int
+}
+
+var seq int
+
+// Basic data struct.
+type Example struct {
+	Sequence int    `json:"id"`
+	Message  string `json:"message"`
+}
+
+func (d *Demo) Send() error {
+	event := cloudevents.Event{
+		Context: d.context(),
+		Data: &Example{
+			Sequence: seq,
+			Message:  d.Message,
+		},
+	}
+	seq++
+	return d.Client.Send(event)
+}
+
+func (d *Demo) context() cloudevents.EventContext {
+	now := time.Now()
+	ctx := cloudevents.EventContextV01{
+		EventID:     uuid.New().String(),
+		EventType:   d.EventType,
+		EventTime:   &types.Timestamp{Time: now},
+		Source:      types.URLRef{URL: d.Source},
+		ContentType: d.ContentType,
+	}
+	return ctx
+}
+
+func _main(args []string, env envConfig) int {
```

Comment:
It would be nice for the main method to have a comment giving an overview of what the sample is trying to demonstrate.

Reply:
I added a todo to add more documentation.